codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/IdempotencyTokenMiddlewareGenerator.java

```diff
@@ -74,23 +105,47 @@ public void writeAdditionalFiles(
             SymbolProvider symbolProvider,
             TriConsumer<String, String, Consumer<GoWriter>> writerFactory
```

Comment:
Can the `writerFactory` parameter should be replaced with the GoDeledgator

Reply:
`writeAdditionalFiles` is a pre-defined GoIntegration hook, and is used at a lot of other places. So probably don't want to change that paramater. Alternate idea is to create a new method for this, but that is redundant as `writeAdditionalFiles` can handle the use case below.